cluster/store/schema.go

```diff
@@ -243,38 +243,52 @@ func (s *schema) addProperty(class string, v uint64, props ...*models.Property)
 
 func (s *schema) addTenants(class string, v uint64, req *command.AddTenantsRequest) error {
 	req.Tenants = removeNilTenants(req.Tenants)
-	s.Lock()
-	defer s.Unlock()
 
+	s.Lock()
 	meta := s.Classes[class]
+	info := s.Classes[class].ClassInfo()
+	s.Unlock()
+
 	if meta == nil {
 		return errClassNotFound
 	}
+	if !info.MultiTenancy.Enabled {
+		return fmt.Errorf("multi-tenancy is not enabled for class %q", class)
+	}
 
-	return meta.AddTenants(s.nodeID, req, v)
+	return meta.AddTenants(s.nodeID, req, int64(info.ReplicationFactor), v)
 }
 
 func (s *schema) deleteTenants(class string, v uint64, req *command.DeleteTenantsRequest) error {
 	s.Lock()
-	defer s.Unlock()
-
+	info := s.Classes[class].ClassInfo()
 	meta := s.Classes[class]
+	s.Unlock()
+
 	if meta == nil {
 		return errClassNotFound
 	}
+	if !info.MultiTenancy.Enabled {
+		return fmt.Errorf("multi-tenancy is not enabled for class %q", class)
+	}
 
 	return meta.DeleteTenants(req, v)
 }
 
 func (s *schema) updateTenants(class string, v uint64, req *command.UpdateTenantsRequest) (n int, err error) {
 	s.Lock()
-	defer s.Unlock()
-
+	info := s.Classes[class].ClassInfo()
 	meta := s.Classes[class]
+	s.Unlock()
+
 	if meta == nil {
 		return 0, errClassNotFound
 	}
 
+	if !info.MultiTenancy.Enabled {
```

Comment:
**suggestion:** we have that pattern in multiple places with comment 
```
s.RLock()
// To avoid races between checking that multi tenancy is enabled and reading from the class itself,
// ensure we fetch both using the same lock.
// We will then read the tenants from the class using the more specific meta lock
info := s.Classes[class].ClassInfo()
meta := s.Classes[class]
s.RUnlock()
```

what do you think if we can wrap it in  func `RLockGuard` etc.

Reply:
also `if !info.MultiTenancy.Enabled {` we can wrap it in func to avoid dup and have a handy method to use